### ISSUE TITLE
Connection reset with large amounts of documents

### ISSUE THREAD

**Issue by TrollWarlord** [rethinkdb/rethinkdb-go#137]
It seems that the more documents you have, the higher chance of a disconnect. Then the lack of any retry policy causes your application to die.

Example query that causes this:

```

func DomainDocument(ctx *rethinkdb.Context) r.Term {
    return r.Db(ctx.Config.Database).Table("domains")
}

func FindDomainByDomain(ctx *rethinkdb.Context, domain string) (*Domain, error) {
    var err error
    var cursor *r.Cursor

    if cursor, err = DomainDocument(ctx).Filter(r.Row.Field("Domain").Eq(domain)).Run(ctx.Session); err != nil {
        return nil, err
    }
    defer cursor.Close()

    if cursor.IsNil() {
        return nil, nil
    }

    var d Domain
    if err = cursor.One(&d); err != nil {
        return nil, err
    }

    return &d, nil
}
```

**Comment by crosbymichael** [comment]
Are you getting `gorethink: use of closed network connection`?

**Comment by dancannon** [comment]
Yes I believe that was the issue that @TrollWarlord was getting.

**Comment by crosbymichael** [comment]
Ok, i am getting the same error.  I'll see if I can debug this more.  It seems like it maybe around the `cursor.Close()` code.

**Comment by dancannon** [comment]
Any help you could provide would be fantastic. Unfortunately I have not had a lot of time to work on this issue lately.

**Comment by dancannon** [comment]
I have completely rewritten the connection pool now and changed how connections are closed and released to the pool (#125).

If you could try your code with the `develop` branch and let me know if you are still getting this error that would be great.

**Comment by dancannon** [comment]
I believe this issue has now been fixed. If you still have issues let me know and ill have another look.
